SimCalorimetry/HGCalAssociatorProducers/plugins/LCToCPAssociatorByEnergyScoreImpl.cc

```diff
@@ -369,6 +372,13 @@ ticl::association LCToCPAssociatorByEnergyScoreImpl<HIT>::makeConnections(
     // It is the inverse of the denominator of the LCToCP score formula. Observe that this is the sum of the squares.
     float invLayerClusterEnergyWeight = 0.f;
     for (auto const& haf : hits_and_fractions) {
+      if constexpr (std::is_same_v<HIT, HGCRecHit>) {
```

Comment:
If the type is `HGCRecHit`, why do you need to check if that hit is in the barrel?
And viceversa as well.